### PR TITLE
Add validated applicative to accumulate errors when validating tests

### PR DESCRIPTION
**Pull request by dariusSalem**

### CODE REVIEW

**Comment by dariusSalem** [comment]
image
Richer errors

**Comment by dariusSalem** [comment]
We could also , in a future commit, decorate our test output with failure details

**Comment by louis-jikkenio** [comment]
> We could also , in a future commit, decorate our test output with failure details

We should already be doing this. When our executor runs it spits out errors as they happen (basically hardcoded into the execution methods), and we have a diff which spits out data discrepancies (all of them).

File: Cargo.toml [hunk 1]
```diff
@@ -48,3 +48,5 @@ hex = { version = "0.4" }
 sha2 = { version = "0.10" }
 sysinfo = { version = "0.29" }
 glob = { version = "0.3.1" }
+nonempty-collections = { version="0.1.5" }
```
Comment (louis-jikkenio) [inline Cargo.toml:51]: Can we split up the cargo.toml to have a [dev-dependencies] section. We can move some of the dependencies (such as the nonempty-collections into that, as those dependencies aren't included in the release build. I'm worried about too many packages bloating the package size.

**Review by louis-jikkenio — CHANGES_REQUESTED** [review]

**Review by louis-jikkenio — APPROVED** [review]

**Pull request merged**